API/ev3_array.h

```diff
@@ -0,0 +1,197 @@
+/** \file ev3_array.h
+ * \brief Dynamically-allocated resizable arrays
+ *
+ * License:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/
+ *
+ * The Initial Developer of this code is Jakub Vanek.
+ * Portions created by Jakub Vanek are Copyright (C) 2020 Jakub Vanek.
+ * All Rights Reserved.
+ *
+ * ----------------------------------------------------------------------------
+ *
+ * \author Jakub Vanek (linuxtardis_at_gmail.com)
+ * \date 2020-03-03
+ * \version 1
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef ev3_array_h
+#define ev3_array_h
+
+////////////////////////////////////////////////////////////////
+// INCLUDES
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+////////////////////////////////////////////////////////////////
+// MACROS
+
+/*!
+ * \brief Initialize an empty resizable array.
+ * \param size Size of each element (sizeof(x)).
+ * \param alignment Alignment requirement of each element (alignof(x)).
+ */
+#define DYNAMIC_CARRAY_INITIALIZER(size, alignment) \
+    ((DynamicCArray) {                             \
+        .elements = NULL,                         \
+        .count    =  0,                           \
+        .capacity =  0,                           \
+        .element_sizeof  = (size),                \
+        .element_alignof = (alignment),           \
```

Comment:
Do we need to store this?

Reply:
Nope, only the stride gets used afterwards. I have removed in now.